File: kfcore/src/main/java/com/lq/ren/base/data/obs/DataObserver.java
package com.lq.ren.base.data.obs;

import androidx.annotation.NonNull;

/**
 * Author lqren on 17/10/11.
 */

public interface DataObserver<Data> {

    void onNext(DataObservable<Data> observable, @NonNull Data data);

    void onError(DataObservable<Data> observable, Throwable throwable);
}

File: kfcore/src/main/java/com/lq/ren/base/data/obs/DataObserver.kt
package com.lq.ren.base.data.obs

/**
 * Author lqren on 17/10/11.
 */

interface DataObserver<Data> {

    fun onNext(observable: DataObservable<Data>, data: Data)

    fun onError(observable: DataObservable<Data>, throwable: Throwable)
}
